internal/internal_task_pollers.go

```diff
@@ -256,9 +252,19 @@ func (bp *basePoller) doPoll(
 	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
 
 	go func() {
+		defer close(doneC)
+		defer func() {
```

Comment:
I'd move panic handling to the top (so it's the first defer entry in the stack) to be future proof. e.g. if the `close(doneC)` may cause panic if it's closed before already.

Reply:
My thought was that we should always have the close be the final statement to ensure that there isn't a race condition. Otherwise we might be changing the result to add the err after the other goroutine is done waiting.

Reply:
I don't see a race condition here. just to make sure we are clear, this panic recovery will not catch panics from the initiator routine.

Reply:
synced offline. we don't need `doneC`, we can just use `ctx.Done()` instead. and the defer order should be
- defer cancel()
- defer func() { recover() ..) }